code/modules/unit_tests/configuration_documentation.dm

```diff
@@ -0,0 +1,63 @@
+/datum/unit_test/configuration_documentation
+	/// List of undocumented config entries
+	var/list/undocumented_entries
+	/// List of set config entries with no code equivalent
+	var/list/extraneous_entries
+	/// List of entries seen already
+	var/list/seen_entries
+
+/datum/unit_test/configuration_documentation/Run()
+	undocumented_entries = list()
+	for(var/I in global.config.entries)
+		var/datum/config_entry/CE = global.config.entries[I]
+		if(!CE.deprecated_by)
+			undocumented_entries += I
+
+	extraneous_entries = list()
+	seen_entries = list()
+	var/test_config_file = world.params["original_config"] ? world.params["original_config"] : DEFAULT_CONFIGURATION_FILE
+	TestGraph(test_config_file)
+	if(undocumented_entries.len)
+		Fail("The following configuration entries are missing default values in the .txt (commented out or otherwise): [english_list(undocumented_entries)]")
+	if(extraneous_entries.len)
```

Comment:
Nothing touches `extraneous_entries`, are you sure this is correct?

Reply:
It's populated in `TryRemoveEntry` if it finds one it doesn't recognize

Reply:
Didn't realize it was a variable on the test :+1: